### PR TITLE
Fix installing Poetry for Python3

### PR DESCRIPTION
**Pull request by hongquan**
Previously, running _get-poetry.py_ with Python 3 didn't give you ability to invoke Poetry with Python 3. This PR is to fix that.

The PR also update installation documentation, giving more examples.

### Pull Request Check List

- [ ] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,14 @@ recommended way of installing `poetry`.
 curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
 ```
 
+or, to install Poetry with Python3:
```
Comment (Jamim) [inline README.md]: Hello @hongquan,
Just a minor suggestion here:
```suggestion
or, to install Poetry with Python 3:
```

**Comment by Jamim** [comment]
Corresponding issue: #1042.

File: get-poetry.py [hunk 2]
```diff
@@ -194,7 +195,7 @@ def expanduser(path):
 POETRY_LIB_BACKUP = os.path.join(POETRY_HOME, "lib-backup")
 
 
-BIN = """#!/usr/bin/env python
+BIN = """#!/usr/bin/env {python_exe}
```
Comment (bersace) [inline get-poetry.py]: I would simple replace BIN with `f"#!{sys.executable}"`. What is the purpose of using env ? Why use project python to run poetry ?

Comment (hongquan) [reply get-poetry.py]: I don't know if replacing `BIN` with `f"#!{sys.executable}"` could break the behavior.

Currently, `poetry install` correctly install the Python packages to pre-selected virtual env. If change to `f"#!{sys.executable}"`, I wonder if that behaviour can still be kept.

Comment (k4ml) [reply get-poetry.py]:
```
> /home/kamal/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/console/commands/install.py(52)handle()
-> extras = []
(Pdb) l
 47  	        installer = Installer(
 48  	            self.io, self.env, self.poetry.package, self.poetry.locker, self.poetry.pool
 49  	        )
 50  	        import pdb;pdb.set_trace()
 51  	
 52  ->	        extras = []
 53  	        for extra in self.option("extras"):
 54  	            if " " in extra:
 55  	                extras += [e.strip() for e in extra.split(" ")]
 56  	            else:
 57  	                extras.append(extra)
(Pdb) self.env
VirtualEnv("/home/kamal/python/demo/.venv")
(Pdb) import sys
(Pdb) sys.executable
'/home/kamal/.local/pipx/venvs/poetry/bin/python'
```
As we can see above, the installer already running with correct venv context, independent of `sys.executable`.

Comment (wohali) [reply get-poetry.py]: Can confirm that this small delta fixes the script:

```diff
--- get-poetry.py       2020-03-25 17:06:28.451064345 -0400
+++ get-poetry.py.new   2020-03-25 17:04:19.746420009 -0400
@@ -197,7 +197,7 @@
 POETRY_LIB_BACKUP = os.path.join(POETRY_HOME, "lib-backup")
 
 
-BIN = """#!/usr/bin/env python
+BIN = f"""#!{sys.executable}
 # -*- coding: utf-8 -*-
 import glob
 import sys
```

along with invoking the script with `python3`, on Debian Buster (10.x).

Comment (hongquan) [reply get-poetry.py]: If so, I opt for `f"#!{sys.executable}"`. The only issue is that `f-string` can only qork with Python 3.6+.

I wonder if @sdispater want to maintain compatibility with Python 3.5. This is the default Python version of Debian Stretch (9).

File: README.md [hunk 3]
```diff
@@ -19,6 +19,14 @@ recommended way of installing `poetry`.
 curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
 ```
 
+or, to install Poetry with Python 3:
```
Comment (NickVolynkin) [inline README.md]: Probably, this should be the first option, as Python 2 is unsupported.

File: README.md [hunk 4]
```diff
@@ -27,6 +35,12 @@ If you want to install prerelease versions, you can do so by passing `--preview`
 python get-poetry.py --preview
```
Comment (NickVolynkin) [inline README.md]: It's absolutely not clear from the docs, where to find this `get-poetry.py`, because `curl` (or `wget`) doesn't save it.

**Comment by NickVolynkin** [comment]
This PR is just great and I'd love to see it merged. Is there anything I could help with, like testing on different systems and Python versions?

File: README.md (hunk 4 above)
Comment (hongquan) [reply README.md]: @NickVolynkin

```console
wget https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py
```

(Replace `curl -sSL` with just `wget`)

will save the install script to file.

File: get-poetry.py (hunk 2 above)
Comment (wohali) [reply get-poetry.py]: Right, in my parallel PR I moved back to a `.format()` approach, see https://github.com/python-poetry/poetry/pull/2222 .

Happy to close mine if this one has traction, but mine's passing all the tests right now.

File: README.md (hunk 4 above)
Comment (NickVolynkin) [reply README.md]: @hongquan Sure, it will. But documentation just supposes that the file is already there. It can confuse readers.

Comment (hongquan) [reply README.md]: @NickVolynkin Thanks. I will update that detail.

**Comment by finswimmer** [comment]
Thanks a lot for your contribution. :+1: 

In the meantime https://github.com/python-poetry/poetry/pull/2547 was created and merged. So I will close this one here.

fin swimmer

**Pull request closed without merging**